### PR TITLE
2019wk12 fix few minor compilation warnings

### PR DESCRIPTION
**Pull request by kerolasa**
This pull request contains five first commits from earlier pull[1] that was requested to be split. That is now done, as well as rebase on top of most recent origin/master.

[1] https://github.com/cronie-crond/cronie/pull/10

### CODE REVIEW

File: anacron/log.c [hunk 1]
```diff
@@ -79,7 +79,7 @@ make_msg(const char *fmt, va_list args)
     /* There's some confusion in the documentation about what vsnprintf
      * returns when the buffer overflows.  Hmmm... */
     len = vsnprintf(msg, sizeof(msg), fmt, args);
-    if (len >= sizeof(msg) - 1)
+    if (0 <= len && (size_t) len >= sizeof(msg) - 1)
```
Comment (t8m) [inline anacron/log.c]: Can you please change it to rather cast the right side to int? And not doing the explicit comparison to zero.

File: anacron/runjob.c [hunk 2]
```diff
@@ -53,7 +53,7 @@ temp_file(job_rec *jr)
 	dir = P_tmpdir;
 
     len = snprintf(template, sizeof(template), "%s/$anacronXXXXXX", dir);
-    if (len >= sizeof(template))
+    if (0 <= len && (size_t) len >= sizeof(template))
```
Comment (t8m) [inline anacron/runjob.c]: Same as above, but maybe also handle the negative len somehow- such as die_е("snprintf error");

File: anacron/log.c (hunk 1 above)
Comment (t8m) [reply anacron/log.c]: Or handle also the negative len as separate case by strcpying some message like "(vnsprintf error)" into msg.

Comment (kerolasa) [reply anacron/log.c]: These `0 <= len` are now done better way in commit 475e869ef303e1eca2d2951253b61b6d03b2395c.

Pull request is also rebased on top of more recent origin/master that fixed a merge conflict.

**Comment by t8m** [comment]
I've merged the patches manually. I've slightly modified the last patch.

**Pull request closed without merging**